rust/lance-core/src/io/local.rs

```diff
@@ -64,38 +64,46 @@ pub struct LocalObjectReader {
 }
 
 impl LocalObjectReader {
-    pub fn open_local_path(
+    pub async fn open_local_path(
         path: impl AsRef<std::path::Path>,
         block_size: usize,
     ) -> Result<Box<dyn Reader>> {
-        let local_file = File::open(&path)?;
-        let object_store_path = Path::from_filesystem_path(path)?;
-        Ok(Box::new(Self {
-            file: Arc::new(local_file),
-            path: object_store_path,
-            block_size,
-        }))
+        let path = path.as_ref().to_owned();
+        let object_store_path = Path::from_filesystem_path(&path)?;
+        tokio::task::spawn_blocking(move || {
+            let local_file = File::open(&path)?;
+            Ok(Box::new(Self {
+                file: Arc::new(local_file),
+                path: object_store_path,
+                block_size,
+            }) as Box<dyn Reader>)
+        })
+        .await?
```

Comment:
Another option could be to use `tokio::fs::File` instead of `std::fs::FIle`.  I'm not aware of any differences

Reply:
CC @wjones127

Reply:
Yeah I think `tokio::fs::File` isn't much different that what you are doing here.